pkg/block/local/adapter.go

```diff
@@ -542,7 +542,17 @@ func (l *Adapter) GetStorageNamespaceInfo() block.StorageNamespaceInfo {
 func (l *Adapter) ResolveNamespace(storageNamespace, key string, identifierType block.IdentifierType) (block.QualifiedKey, error) {
 	qk, err := block.DefaultResolveNamespace(storageNamespace, key, identifierType)
 	if err != nil {
-		return qk, err
+		return nil, err
+	}
+
+	// Check if path allowed and return error if path is not allowed
+	_, err = l.extractParamsFromObj(block.ObjectPointer{
```

Comment:
Are we sure that's the appropriate location to check it? It seems weird to have a `ResolveNamespace` method checking permissions using `extract..` . Maybe we should use other names for the `extract` function?

Reply:
I think I'll just *extract* the path validation from the extract method 😃

Reply:
Tried to separate the logic, but it doesn't feel natural. Prefer to leave it as is - I think this is the lesser evil 😄